app/components/ConfirmTransaction/index.js

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import HdKey from 'ethereumjs-wallet/hdkey'
+import Bip39 from 'bip39'
+import CryptoJs from 'crypto-js'
+
+import 'components/App/styles.css'
+import { assignTransaction } from '../../actions/transactions'
+
+class ConfirmTransaction extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      password: '',
+      errorMessage: '',
+      transaction: {},
+    }
+
+    chrome.windows.getCurrent((window) => {
+      this.showTransaction(window.id)
+    })
+  }
+
+  showTransaction = (windowId) => {
+    const { transactions } = this.props
+
+    if (transactions && transactions.length > 0) {
+      const ts = transactions.filter(t => t.popupId === windowId)
+
+      if (ts.length === 1 && ts[0].tx) {
+        const tx = ts[0].tx
+
+        this.setState({
```

Comment:
Calling setState from the constructor is not allowed, why not just showTransaction returns the transaction and you specify the ```this.state = ... ``` inside... ```getCurrent``` function?

Reply:
`getCurrent` function is asyncronous so I'm moving it to `componentDidMount`

Reply:
That makes sense.